src/Build/Evaluation/Expander.cs

```diff
@@ -2012,6 +2061,23 @@ internal static bool ExpandExpressionCapture<S>(
                 // If there are no items of the given type, then bail out early
                 if (itemsOfType.Count == 0)
                 {
+                    if (expressionCapture.Captures?.Any(capture =>
+                        {
+                            if (string.Equals(capture.FunctionName, "Count", StringComparison.OrdinalIgnoreCase))
+                            {
+                                return true;
+                            }
+                            if (string.Equals(capture.FunctionName, "AnyHaveMetadataValue", StringComparison.OrdinalIgnoreCase))
+                            {
+                                return true;
+                            }
+                            return false;
+                        }) != true)
```

Comment:
Is it this?

```suggestion
                    if (expressionCapture.Captures?.Any(capture =>
                        !string.Equals(capture.FunctionName, "Count", StringComparison.OrdinalIgnoreCase) ||
                        !string.Equals(capture.FunctionName, "AnyHaveMetadataValue", StringComparison.OrdinalIgnoreCase)
                        )
```

Reply:
My brain is kind of fried at the moment, so I pasted this into .netfiddle:
```
using System;
using System.Collections.Generic;
using System.Linq;
					
public class Program
{
	public static void Main()
	{
		 var x = new List<string>();
		 x.Add("Count");
		 if (x.Any(capture =>
                        {
                            if (string.Equals(capture, "Count", StringComparison.OrdinalIgnoreCase))
                            {
                                return true;
                            }
                            if (string.Equals(capture, "AnyHaveMetadataValue", StringComparison.OrdinalIgnoreCase))
                            {
                                return true;
                            }
                            return false;
                        }) != true)
		 {
		 	Console.WriteLine("hit1");
		 }
	    if (x.Any(capture =>
												 !string.Equals(capture, "Count", StringComparison.OrdinalIgnoreCase) ||
												 !string.Equals(capture, "AnyHaveMetadataValue", StringComparison.OrdinalIgnoreCase)))
			{
			Console.WriteLine("hit2");
			}
	}
}
```
These aren't equivalent - empty collection hits only the first case, collection with "Count" hits only the second one.
I'll look at it closer when refactoring, some simplification along these lines is definitely possible, but I found myself repeatedly getting lost in the logic so in the end I simplified at the cost of verbosity.

Reply:
Yuliia suggested a similar change with a similar simplification. In the end I went with that one.